### PR TITLE
Defrag cleanup/v6

### PR DESCRIPTION
**Pull request by victorjulien**
bring defrag hash handling closer to flow hash handling:
- free trackers on lookup
- new stats
 
Also modified `max-frags` behavior: if we hit max-frags the tracker is now freed.  

replace #11141, rebasing

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20903

File: etc/schema.json [hunk 1]
```diff
@@ -5184,8 +5190,29 @@
...
+                        "tracker_soft_reuse": {
+                            "type": "integer",
+                            "description":
+                                    "How many times a tracker was force reused after it was complete"
+                        },
+                        "tracker_hard_reuse": {
+                            "type": "integer",
+                            "description":
+                                    "How many times a tracker was force reused before it was done"
+                        },
```
Comment (jasonish) [inline etc/schema.json:5202]: I'd like further explanation of what these really mean? If only in the commit.

Comment (jasonish) [inline etc/schema.json:5202]: I'd like further explanation of what these really mean? If only in the commit.

**Review by jasonish — CHANGES_REQUESTED** [review]

File: etc/schema.json (hunk 1 above)
Comment (victorjulien) [reply etc/schema.json:5202]: Open to better naming.

"tracker_soft_reuse" spare pool was empty and memcap is reached, so the packet thread that needed a tracker went into hash directly to reuse a tracker from the hash, and it found one that was already marked `remove` (so reassembly finished).

"tracker_hard_reuse" spare pool was empty and the memcap is reached, so the packet thread that needed a tracker went into hash directly to reuse a tracker from the hash, but it found one that was still in use (`remove` false) and reused that. The original tracker is cleared leading to the packet for that tracker not being reassembled.

Comment (jasonish) [reply etc/schema.json:5202]: Maybe description updates:

### tracker_soft_reuse

Finished tracker re-used from hash table before being moved to spare pool.

### tracker_hard_reuse

Active tracker force closed before completion and reused for new tracker.

Comment (victorjulien) [reply etc/schema.json:5202]: see https://github.com/OISF/suricata/pull/11224

**Comment by victorjulien** [comment]
replaced by #11224

**Pull request closed without merging**